components/automate-cli/pkg/verifyserver/services/fqdnservice/fqdnservice_test.go

```diff
@@ -0,0 +1,707 @@
+						SuccessMsg:    constants.FQDN_SUCCESS_MESSAGE,
+						ErrorMsg:      "",
+						ResolutionMsg: "",
+					},
+					{
+						Title:         constants.NODE_TITLE,
+						Passed:        true,
+						SuccessMsg:    constants.NODE_SUCCESS_MESSAGE,
+						ErrorMsg:      "",
+						ResolutionMsg: "",
+					},
+					{
+						Title:         constants.CERTIFICATE_TITLE,
+						Passed:        true,
+						SuccessMsg:    constants.CERTIFICATE_SUCCESS_MESSAGE,
+						ErrorMsg:      "",
+						ResolutionMsg: "",
+					},
+				},
+			},
+			httpsSuccessPort,
+		},
+		{
+			"Case is Before Deployment, FQDN and Nodes are not reachable, Certicate is valid",
+			models.FqdnRequest{
+				Fqdn:              LOCALHOST2,
+				NodeType:          constants.CHEF_INFRA_SERVER,
+				RootCert:          CA_CERT,
+				IsAfterDeployment: false,
+				Nodes: []string{
+					"172.154.0.2",
+				},
+			},
+			models.FqdnResponse{
+				Passed: false,
+				Checks: []models.Checks{
+					{
+						Title:         constants.FQDN_TITLE,
+						Passed:        false,
+						SuccessMsg:    "",
+						ErrorMsg:      constants.FQDN_ERROR_MESSAGE,
+						ResolutionMsg: constants.FQDN_RESOLUTION_MESSAGE,
+					},
+					{
+						Title:         constants.NODE_TITLE,
+						Passed:        false,
+						SuccessMsg:    "",
+						ErrorMsg:      fmt.Sprintf(constants.NODE_ERROR_MESSAGE, "[172.154.0.2]"),
+						ResolutionMsg: constants.NODE_RESOLUTION_MESSAGE,
+					},
+					{
+						Title:         constants.CERTIFICATE_TITLE,
+						Passed:        true,
+						SuccessMsg:    constants.CERTIFICATE_SUCCESS_MESSAGE,
+						ErrorMsg:      "",
+						ResolutionMsg: "",
+					},
+				},
+			},
+			httpsSuccessPort,
+		},
+		{
+			"Case is Before Deployment, FQDN and Nodes are not reachable, Certificate block type is wrong",
+			models.FqdnRequest{
+				Fqdn:              LOCALHOST,
+				NodeType:          constants.CHEF_INFRA_SERVER,
+				RootCert:          INVALID_BLOCK_CA_CERT,
+				IsAfterDeployment: false,
+				Nodes: []string{
+					"172.154.0.2",
+				},
+			},
+			models.FqdnResponse{
+				Passed: false,
+				Checks: []models.Checks{
+					{
+						Title:         constants.FQDN_TITLE,
+						Passed:        false,
+						SuccessMsg:    "",
+						ErrorMsg:      constants.FQDN_ERROR_MESSAGE,
+						ResolutionMsg: constants.FQDN_RESOLUTION_MESSAGE,
+					},
+					{
+						Title:         constants.NODE_TITLE,
+						Passed:        false,
+						SuccessMsg:    "",
+						ErrorMsg:      fmt.Sprintf(constants.NODE_ERROR_MESSAGE, "[172.154.0.2]"),
+						ResolutionMsg: constants.NODE_RESOLUTION_MESSAGE,
+					},
+					{
+						Title:         constants.CERTIFICATE_TITLE,
+						Passed:        false,
+						SuccessMsg:    "",
+						ErrorMsg:      constants.CERTIFICATE_ERROR_MESSAGE,
+						ResolutionMsg: constants.CERTIFICATE_RESOLUTION_MESSAGE,
+					},
+				},
+			},
+			httpsSuccessPort,
+		},
+		{
+			"Case is Before Deployment, FQDN and Nodes are not reachable, Certicate is not SAN based",
```

Comment:
Can we have a case, where FQDN is reachable but nodes are not?

Reply:
done